teuthology/test/test_scrape.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import with_statement
+
+import os
+import shutil
+import tempfile
+import yaml
+from teuthology import scrape
+
+class FakeResultDir(object):
+    """Mocks a Result Directory"""
+
+    def __init__(self,
+                failure_reason="Dummy reason",
+                assertion="FAILED assert 1 == 2\n",
+                blank_backtrace=False
+    ):
+        self.failure_reason = failure_reason
+        self.assertion = assertion
+        self.blank_backtrace = blank_backtrace
+        self.path = tempfile.mkdtemp()
+        
+        with open(os.path.join(self.path, "config.yaml"), "w") as f:
+            yaml.dump({"description": "Dummy test"}, f)
+        
+        with open(os.path.join(self.path, "summary.yaml"), "w") as f:
+            yaml.dump({
+                "success": "false",
+                "failure_reason": self.failure_reason
+            }, f)
+
+        with open(os.path.join(self.path, "teuthology.log"), "w") as f:
+            if not self.blank_backtrace:
+                f.write(" ceph version 1000\n")
+                f.write(".stderr: Dummy error\n")
+                f.write(self.assertion)
+            f.write(" NOTE: a copy of the executable dummy text\n")
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_typ, exc_val, exc_tb):
+        shutil.rmtree(self.path)
+
+class TestScrape(object):
+    """Tests for teuthology.scrape"""
+
+    def test_grep(self):
+        with open("/tmp/scrapetest.txt", 'w') as f:
```

Comment:
The hardcoded file name is not thread safe, could you please use `tempfile` for creating temporary files?

Reply:
Sure. Doing this right away!

Reply:
Actually `grep` takes a filename as a parameter. With `tempfile` I don't think you get a valid path.
So, I am thinking of using something like `uuid` as the filename.

Reply:
Used the `FakeResultDir` class I defined in the file.